### PR TITLE
multi: update to fn v2

### PR DESCRIPTION
**Pull request by ProofOfKeags**
## Change Description
Updates the `tlv` package to use `fn` v2. This preps for a larger upgrade.

## Steps to Test
`make unit`

### Testing
- [ ] Your PR passes all CI checks.
- [ ] Tests covering the positive and negative (error paths) are included.
- [ ] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [ ] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [ ] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [ ] Any new logging statements use an appropriate subsystem and logging level.
- [ ] Any new lncli commands have appropriate tags in the comments for the rpc in the proto file.
- [ ] [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

📝 Please see our [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.

### CODE REVIEW

File: protofsm/state_machine.go [hunk 1]
```diff
@@ -346,17 +348,31 @@ func (s *StateMachine[Event, Env]) executeDaemonEvent(
 
 			// If a post-send event was specified, then we'll funnel
 			// that back into the main state machine now as well.
-			return fn.MapOptionZ(daemonEvent.PostSendEvent, func(event Event) error { //nolint:lll
-				return s.wg.Go(func(ctx context.Context) {
-					log.Debugf("FSM(%v): sending "+
-						"post-send event: %v",
-						s.cfg.Env.Name(),
-						lnutils.SpewLogClosure(event),
+			ok := fn.ElimOption(
```
Comment (guggero) [inline protofsm/state_machine.go:351]: Why the switch to `ElimOption` here with the `func() bool { return true },` function? Can't this just be `MapOptionZ` with a boolean returned instead?

File: protofsm/state_machine.go [hunk 2]
```diff
@@ -346,17 +348,31 @@ func (s *StateMachine[Event, Env]) executeDaemonEvent(
...
+					return s.wg.Go(
+						func(ctx context.Context) {
+							log()
+							s.SendEvent(event)
+						},
 					)
+				},
+			)
+			if !ok {
+				return fmt.Errorf("StateMachine is stopping.")
```
Comment (guggero) [inline protofsm/state_machine.go:372]: Linter not happy about the punctuation in the error message. Here and below.

File: sweep/tx_input_set.go [hunk 3]
```diff
@@ -141,22 +141,28 @@ func validateInputs(inputs []SweeperInput, deadlineHeight int32) error {
...
-				}
-
-				// Update the deadline height if it's
-				// different.
-				inputDeadline = h
-			})
-
-			return inp.OutPoint()
-		}, inputs)...,
+		fn.Map(
```
Comment (guggero) [inline sweep/tx_input_set.go:144]: I'm glad we changed the semantics before we're using `fn` everywhere. That diff wasn't too hard to get through :+1:

**Review by guggero — COMMENTED** [review]
CI not happy with the second commit. I think we have to squash commit 2 and 3 together to make things compile.
LGTM after CI is green :tada:

File: protofsm/state_machine.go (hunk 1 above)
Comment (guggero) [reply protofsm/state_machine.go:351]: Ah, right, makes sense :+1:

**Comment by guggero** [comment]
Pushed a new tag `tlv/v1.3.0`, can be rebased.

**Comment by ProofOfKeags** [comment]
This has so many rebase conflicts that I think it just makes more sense to redo the commit / PR

**Pull request closed without merging**